src/jdk.compiler/share/classes/com/sun/tools/javac/comp/LambdaToMethod.java

```diff
@@ -755,6 +755,12 @@ private void addDeserializationCase(MethodHandleSymbol refSym, Type targetType,
         String functionalInterfaceClass = classSig(targetType);
         String functionalInterfaceMethodName = samSym.getSimpleName().toString();
         String functionalInterfaceMethodSignature = typeSig(types.erasure(samSym.type));
+        if ((refSym.flags_field & OBJECT_METHOD_IN_INTERFACE) != 0) {
```

Comment:
Instead of using a new flag, can we use the base symbol and check if its owner is Object?

Reply:
Thanks for the comment. Yes, that is possible:
https://github.com/openjdk/jdk/pull/8054/commits/7bf7fedbf122717bdcc13d5ec4fc821590b8e82a